pkg/minikube/cruntime/containerd.go

```diff
@@ -188,10 +189,12 @@ func generateContainerdConfig(cr CommandRunner, imageRepository string, kv semve
 		PodInfraContainerImage string
 		SystemdCgroup          bool
 		InsecureRegistry       []string
+		CNIConfDir             string
 	}{
 		PodInfraContainerImage: pauseImage,
 		SystemdCgroup:          forceSystemd,
 		InsecureRegistry:       insecureRegistry,
+		CNIConfDir:             cni.CNIConfDir,
```

Comment:
how about we add this to the CRIO too ?

Reply:
yes, i thought about that too, just wanted to confirm that it will works for containerd first

Reply:
added that one too (below)

Reply:
Actually you will have to add it to cri-dockerd as well, eventually. If we insist on moving it.

`      --cni-conf-dir string                     <Warning: Alpha feature> The full path of the directory in which to search for CNI config files`

Reply:
@afbjorklund thank you for pointing that out

it seems that all three runtimes behave a bit differently towards that requirement for custom cni-conf-dir and i'm not sure if docker _requires_ it atm (but i might be wrong)

eg, _crio_ is quite fine but _containerd_ would complain [re]starting with an empty cni-conf-dir ('"lastCNILoadStatus": "cni config load failed: no network config found in /etc/cni/net.mk: cni plugin not initialized: failed to load cni config"') but will work after cni is initialised, thought pods that rely on it could end up in _unknown_ status after the node/cluster restart

that's why i've changed the logic a bit: a runtime would start with the default cni-conf-dir, and then, after the cni is applied, (just) the _containerd_ and _crio_ (for now?) would point to custom cni-conf-dir to pick it up
also
i've changed _where_ we wait, and specifically after the node/cluster restart, it will be _after_ the cni is applied, not before (like we did, and that was the source of many timeouts)

all in all, i think all that gave decent results in terms of tests (eg, Docker_Linux_containerd had 0/102 and Docker_Linux_crio had 1/102 fails, but Docker_Linux 7/250 which is a regression to what it was before - if it's not flaking...)

so, i'm still looking at logs atm to see if there's anything missing, but do let me know if you think docker has to follow the same 'logic', as with this 'restructuring, it's rather easy now to 'implement' that as well